packages/compiler/src/core/checker.ts

```diff
@@ -947,7 +947,21 @@ export function createChecker(program: Program, resolver: NameResolver): Checker
         );
       }
     }
-    return mapper ? mapper.getMappedType(type) : type;
+    if (mapper) {
+      let mappedType = mapper.getMappedType(type);
+      // recursively resolve the mapped type so long as it is
+      // still a TemplateParameter, since there may be multiple
+      // levels of mapping (e.g. when referencing defaults)
+      while (
```

Comment:
haven't look too much into this but wondering if the fix shouldn't be somewhere else.


What is the difference with this code where you reference other template params
```
model T<A, B =A, C = B> {
  a: A,
  b: B,
  c: C,
}


model Test {
  t1: T<string>;
}
```

https://typespec.io/playground/?c=bW9kZWwgVDxBLCBCID1BLCBDID0gQj4gewogIGE6IEEsCiAgYjogQsQIYzogQywKfQoKCsc2ZXN0xSl0MTogVDxzdHJpbmc%2BOwp9&e=%40typespec%2Fopenapi3&options=%7B%7D

Reply:
The above model ends up being treated the same before and after this PR. I only saw this impacting interfaces/operations, when an operation template references a template parameter from the interface as a default value. 

In that scenario, the template mapper had the interface template arguments so there's a mapping from the default TemplateParameter to the interface-level template argument.

I did originally have similar logic in `checkOperation` instead, but this seemed like a simpler place (and less likely to miss cases) to have this with the assumption it's safe to recursively grab the mapped types.

Reply:
I’m maybe more wondering if they shouldn’t be solved even more upstream like when we resolve the default value. How was the example above working before?

Reply:
Hmm, I'll have to dig a bit more - from what I can tell in your example each template parameter that has a default value chains well (so `C` has a default of `B` which has a default of `A`). Maybe because everything is contained in one instantiation, A is resolved to `string`, so the defaults ultimately resolve to that as well.

When I look at the returnType of `readFoo` from my changelog example, it ends up being the TemplateParameter `T`. So it does know that it should use the default value of `T`, but doesn't currently make the jump to getting the mapped type for T (which comes from the interface). It doesn't have a `T` template parameter to resolve - it comes from the parent.